src/models/test/StyleSheetManager.test.js

```diff
@@ -116,6 +119,69 @@ describe('StyleSheetManager', () => {
     expect(document.body.innerHTML).toMatchSnapshot()
   })
 
+  // https://github.com/styled-components/styled-components/issues/1634
+  it('should flush styles to an iframe when the iframe is re-rendered', async () => {
```

Comment:
This test is extremely specific to your environment and additionally this conflicts with it as well:

> Changing the StyleSheet while the component is mounted — essentially hot swapping the sheet — is not something we should support,

Also it's a little unnecessarily complex. Instead we can use enzyme to swap out the the stylesheet itself together with a prop, and expect to see a different class.

Edit: I'd say it's ok to not test this for now. I can write a test eventually, but one that is specific to all caching guarantees instead